lib/src/support/draw_utils.cpp

```diff
@@ -79,7 +79,26 @@ namespace cycfi { namespace elements
       }
    }
 
-   void draw_button(canvas& cnv, rect bounds, color c, bool enabled, float corner_radius)
+   void draw_round_rect(canvas& cnv, rect bounds, std::array<float, 4> radius)
+   {
+      auto l = bounds.left;
+      auto t = bounds.top;
+      auto r = bounds.right;
+      auto b = bounds.bottom;
+      for (auto& r : radius) {
+         r = std::min(r, std::min(bounds.width(), bounds.height()) / 2);
+      }
+
+      // radii array follow the unit circle defined counterclockwise with 0 at (x, y) = (1, 0)
```

Comment:
Isn't it more natural for the radii to follow a clockwise path? This follows the natural API of `cnv.arc(...)` as well, which defaults to a clockwise path.

Reply:
Personally I find the anti-clockwise path _much_ more natural since that's how the unit circle is defined. I was actually very surprised to learn this wasn't the standard in the API. 

I don't have a strong preference if you prefer the other way around though.

Reply:
Perhaps we should just add a small struct with named fields to avoid future confusion.

Reply:
> Perhaps we should just add a small struct with named fields to avoid future confusion.

This, I think, a good solution. However, when passing in a struct the usual way using brace initializer, you will still pass it in sequential order. We still have to choose the right sequence.

Reply:
That's true. I think with a constructor illustrating the order it'll be less important though. 

Anyway, my main argument for the current implementation is that it follows the unit circle, which in turn is the conventional way to define a spherical coordinate system. Again I gotta say I am extremely surprised the W3C API does not follow this convention (I think this is perhaps because the y-axis is flipped?).

On the other hand, perhaps the connection to spherical coordinates is too vague for this connection to hold. I am thinking of this problem in terms of the four arcs we have to draw, so it is an obvious connection for me to make.

Reply:
> I don't have a strong preference if you prefer the other way around though.

After thinking about it some more, I think I'd go with a clockwise direction if that's OK with you.